lib/wasix/src/state/linker.rs

```diff
@@ -2602,14 +2629,21 @@ impl InstanceGroupState {
             if import.module() == "env" {
                 match import.name() {
                     "memory" => {
-                        if !matches!(import.ty(), ExternType::Memory(_)) {
+                        let ExternType::Memory(memory_ty) = import.ty() else {
                             return Err(LinkError::BadImport(
                                 import.module().to_string(),
                                 import.name().to_string(),
                                 import.ty().clone(),
                             ));
-                        }
+                        };
                         trace!(?module_handle, ?import, "Main memory");
+
+                        // Make sure the memory is big enough for the module being instantiated
+                        let current_size = self.memory.grow(store, 0)?;
```

Comment:
I'm not sure calling grow to get the current size is the good way, which implies we are missing some important APIs.

Can we have a method to have the current size of a memory `.size(&store)` and use it here? (in the `api/` crate)